typescript/db/src/repository/upload-s3.ts

```diff
@@ -1,64 +1,102 @@
-// import { S3Client, PutObjectCommand } from "@aws-sdk/client-s3";
-// import { getSignedUrl } from "@aws-sdk/s3-request-presigner";
-// import "isomorphic-fetch";
+import { S3Client, PutObjectCommand } from "@aws-sdk/client-s3";
+import { getSignedUrl } from "@aws-sdk/s3-request-presigner";
+import "isomorphic-fetch";
+import memoizeOne from "memoize-one";
+import { Repository } from "../../../common-resolvers/src";
+import { UploadTargetHttp } from "../../../graphql-types/src/graphql-types.generated";
 
-// import { Repository } from "../../../common-resolvers/src";
-// import { UploadTargetHttp } from "../../../graphql-types/src/graphql-types.generated";
+const bucket = "labelflow";
+const region = process.env?.LABELFLOW_AWS_REGION!;
 
-// const bucket = "labelflow-images";
-// const region = "eu-west-1";
-// const location = `https://${bucket}.s3.${region}.amazonaws.com`;
-// const s3Client = new S3Client({
-//   region,
-//   credentials: {
-//     accessKeyId: process.env?.LABELFLOW_AWS_ACCESS_KEY_ID!,
-//     secretAccessKey: process.env?.LABELFLOW_AWS_SECRET_ACCESS_KEY!,
-//   },
-// });
+const getClient = memoizeOne(
+  () =>
+    new S3Client({
+      ...{
+        region,
+        credentials: {
+          accessKeyId: process.env?.LABELFLOW_AWS_ACCESS_KEY_ID!,
+          secretAccessKey: process.env?.LABELFLOW_AWS_SECRET_ACCESS_KEY!,
+        },
+      },
+      // Uncomment bellow to try making minio work with s3 sdk
+      // See https://docs.min.io/docs/how-to-use-aws-sdk-for-javascript-with-minio-server.html
+      // // If we set an given endpoint, it means that we test with minio
+      // ...(process.env?.LABELFLOW_AWS_ENDPOINT
+      //   ? {
+      //       endpoint: process.env?.LABELFLOW_AWS_ENDPOINT,
+      //       forcePathStyle: true, // required to make minio work
+      //     }
+      //   : {}),
+    })
+);
 
-// export const getUploadTargetHttp = async (
-//   key: string
-// ): Promise<UploadTargetHttp> => {
-//   const command = new PutObjectCommand({
-//     Bucket: bucket,
-//     Key: key,
-//   });
-//   const signedUrl = await getSignedUrl(s3Client, command, {
-//     expiresIn: 3600,
-//   });
-//   return {
-//     __typename: "UploadTargetHttp",
-//     uploadUrl: signedUrl,
-//     downloadUrl: `${location}/${key}`,
-//   };
-// };
+export const uploadsRoute = "/api/downloads";
```

Comment:
This is only used for download so I think it'd be better to rename it to `downloadsRoute`